src/ice_spi.c

```diff
@@ -55,42 +60,29 @@ void ice_spi_init(void) {
     gpio_set_dir(ICE_SPI_SCK_PIN, GPIO_IN);
     gpio_set_dir(ICE_SPI_TX_PIN, GPIO_IN);
     gpio_set_dir(ICE_SPI_RX_PIN, GPIO_IN);
+
+    // Initialize SPI, but don't yet assign the pins SPI function so they stay in high impedance mode.
+    spi_init(spi1, 50 * 1000 * 1000);
+
+    // Setup DMA channel and interrupt handler
+    dma_tx = dma_claim_unused_channel(true);
+    dma_rx = dma_claim_unused_channel(true);
+
+    dma_channel_set_irq1_enabled(dma_rx, true);
```

Comment:
I suppose you chose `irq1` for leaving `irq0` available to the user?
This makes sense as we are providing an SDK/library instead of an application.

Reply:
You can ignore `ice_comm.h`, which was pending to be replaced by `ice_spi.h`, which did take some if its code.

Reply:
Ah, I had missed `ice_comm`!  Looks like that's superseded by this version of `ice_spi` though.

Reply:
ice_comm was only there in case some code of it was not merged into `ice_spi.h` yet, I do not think anything used it.